File: container-with-most-water.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution
{

public:
    
    int ptr_left, ptr_right, max;

    int calculate_area(int left, int right, vector<int>* height)
    {
        return min(height.at(left), height->at(right)) * (ptr_right - ptr_left);
    }

    int maxArea(vector<int>& height)
    {
        ptr_left = 0;
        ptr_right = height.size() - 1;

        max = 0;

        while (ptr_left < ptr_right)
        {
            if (max < calculate_area(&height))
        }
    }

};

int main()
{
    Solution* solution = new Solution();

    vector<int> height {1, 8, 6, 2, 5, 4, 8, 3, 7};

    cout << solution->maxArea(height) << endl;

    return 0;
}

File: container-with-most-water.go
package main

import "fmt"

func min(a int, b int) int {

	if a < b {
		return a
	} else {
		return b
	}

}

func maxArea(height []int) int {
	max := 0

	for i := 0; i < len(height)-1; i++ {

		if (len(height)-1-i)*height[i] > max {
			for j := i + 1; j < len(height); j++ {

				area := min(height[i], height[j]) * (j - i)

				if area > max {
					max = area
				}

			}
		}

	}

	return max
}

func main() {
	container := []int{1, 8, 6, 2, 5, 4, 8, 3, 7}

	fmt.Println(maxArea(container))
}
